bundles/org.openhab.binding.intesis/src/main/java/org/openhab/binding/intesis/internal/handler/IntesisHomeHandler.java

```diff
@@ -0,0 +1,276 @@
+                                sessionId = auth.sessionID;
+                                if (!sessionId.isEmpty()) {
+                                    logger.trace("sessionID : {}", sessionId);
+                                    updateStatus(ThingStatus.ONLINE);
+                                } else {
+                                    updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR);
+                                }
+                                refreshJob = scheduler.scheduleWithFixedDelay(this::getAllUidValues, 0, refreshInterval,
+                                        TimeUnit.SECONDS);
+                            } else {
+                                updateStatus(ThingStatus.OFFLINE);
+                            }
+                        }
+                    } else {
+                        updateStatus(ThingStatus.OFFLINE);
+                    }
+                }
+            } catch (Exception e) {
+            }
+        }, 2, TimeUnit.SECONDS);
+    }
+
+    @Override
+    public void dispose() {
+        logger.debug("IntesisHomeHandler disposed.");
+        final ScheduledFuture<?> refreshJob = this.refreshJob;
+
+        if (refreshJob != null) {
+            refreshJob.cancel(true);
+            this.refreshJob = null;
+        }
+        try {
+            String contentString = "{\"command\":\"logout\",\"data\":{\"sessionID\":\"" + sessionId + "\"}}";
+            api.postRequest(ipAddress, contentString);
+        } catch (Exception e) {
+        }
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        int uid = 0;
+        int value = 0;
+        String channelId = channelUID.getId();
+        if (command instanceof RefreshType) {
+            // The thing is updated by the scheduled automatic refresh so do nothing here.
+        } else {
+            switch (channelId) {
+                case POWER_CHANNEL:
+                    uid = 1;
+                    value = command.equals(OnOffType.OFF) ? 0 : 1;
+                    break;
+                case MODE_CHANNEL:
+                    uid = 2;
+                    value = Integer.parseInt(command.toString());
+                    break;
+                case WINDSPEED_CHANNEL:
+                    uid = 4;
+                    value = Integer.parseInt(command.toString());
+                    break;
+                case SWINGUD_CHANNEL:
+                    uid = 5;
+                    value = Integer.parseInt(command.toString());
+                    break;
+                case SWINGLR_CHANNEL:
+                    uid = 6;
+                    value = Integer.parseInt(command.toString());
+                    break;
+                case TEMP_CHANNEL:
+                    uid = 9;
+                    value = (Integer.parseInt(command.toString().replace(" °C", ""))) * 10;
```

Comment:
As you use Units of Measure, you can cast this to `QuantityType` and use `toUnit()` to convert it to celsius. Your code would fail if the user uses Fahrenheit.

Reply:
No, the command has to deliver just an integer, so I have to remove the unit symvol and multiply the interger value by 10.

Reply:
```suggestion
                    if (command instanceof QuantityType) {
                        value = ((QuantityType) command).toUnit(SIUnits.CELSIUS).intValue()  * 10;
                    }
```